fastfield_codecs/src/multilinearinterpol_2.rs

```diff
@@ -0,0 +1,353 @@
+/*!
+
+MultiLinearInterpol compressor uses linear interpolation to guess a values and stores the offset, but in blocks of 512.
+
+With a CHUNK_SIZE of 512 and 29 byte metadata per block, we get a overhead for metadata of 232 / 512 = 0,45 bits per element.
+The additional space required per element in a block is the the maximum deviation of the linear interpolation estimation function.
+
+E.g. if the maximum deviation of an element is 12, all elements cost 4bits.
+
+Size per block:
+Num Elements * Maximum Deviation from Interpolation + 29 Byte Metadata
+
+*/
+
+use crate::FastFieldCodecReader;
+use crate::FastFieldCodecSerializer;
+use crate::FastFieldDataAccess;
+use crate::FastFieldStats;
+use std::io::{self, Read, Write};
+use std::ops::Sub;
+use tantivy_bitpacker::compute_num_bits;
+use tantivy_bitpacker::BitPacker;
+
+use common::BinarySerializable;
+use common::DeserializeFrom;
+use tantivy_bitpacker::BitUnpacker;
+
+const BLOCK_SIZE: u64 = 512;
+
+#[derive(Clone)]
+pub struct MultiLinearInterpol2FastFieldReader {
+    num_vals: u64,
+    min_value: u64,
+    max_value: u64,
+    block_metadatas: Vec<BlockMetadata>,
+    block_start_offsets: Vec<u64>,
+    block_bit_unpackers: Vec<BitUnpacker>,
+}
+
+#[derive(Clone, Debug, Default)]
+struct BlockMetadata {
+    constant: u64,
+    constant_positive_offset: u64,
+    slope: f32,
+    num_bits: u8,
+}
+
+
+impl BinarySerializable for BlockMetadata {
+    fn serialize<W: Write>(&self, write: &mut W) -> io::Result<()> {
+        self.constant.serialize(write)?;
+        self.constant_positive_offset.serialize(write)?;
+        self.slope.serialize(write)?;
+        self.num_bits.serialize(write)?;
```

Comment:
actually we could get the max vals and apply bitpacking to save some more bits (except from the f32)

(store the num bits in the codec header)

Reply:
I won't do that for now :)